php-classes/Slate/CBL/Tasks/StudentTasksRequestHandler.php

```diff
@@ -162,75 +165,46 @@ public static function handleStudentTaskSubmissionRequest()
-                return $s->ID;
-            }, $originalAttachments);
 
-            $failed = [];
-            $attachmentIds = [];
-            $attachments = [];
-            $defaultAttachmentClass = AbstractTaskAttachment::$defaultClass;
+        $defaultAttachmentClass = AbstractTaskAttachment::class;
 
-            foreach ($data['Attachments'] as $attachmentData) {
+        if (isset($requestData['Attachments'])) {
+            foreach ($requestData['Attachments'] as $attachmentData) {
                 $attachmentClass = $attachmentData['Class'] ?: $defaultAttachmentClass;
                 if ($attachmentData['ID'] >= 1) {
                     if (!$Attachment = $attachmentClass::getByID($attachmentData['ID'])) {
                         $failed[] = $attachmentData;
                         continue;
                     }
+
+                    if (!empty($attachmentData['Status']) && in_array($attachmentData['Status'], $defaultAttachmentClass::getFieldOptions('Status', 'values'))) {
+                         $Attachment->Status = $attachmentData['Status'];
+                     }
                 } else {
                     $Attachment = $attachmentClass::create($attachmentData);
                 }
 
-                $Attachment->ContextID = $Record->ID;
-                $Attachment->ContextClass = $Record->getRootClass();
-                $Attachment->save();
-                $attachments[] = $Attachment;
-                $attachmentIds[] = $Attachment->ID;
-            }
-
-            \DB::nonQuery('DELETE FROM `%s` WHERE ContextClass = "%s" AND ContextID = %u AND ID NOT IN ("%s")', [
-                AbstractTaskAttachment::$tableName,
-                \DB::escape($Record->getRootClass()),
-                $Record->ID,
-                join('", "', $attachmentIds)
-            ]);
-
-        }
-
-        $Record->Attachments = $attachments;
-
-        // update skills
-        if (isset($data['SkillIDs'])) {
-            $originalSkills = $Record->Skills;
-            $originalSkillIds = array_map(function($s) {
-                return $s->ID;
-            }, $originalSkills);
-
-            $oldSkillIds = array_diff($originalSkillIds, $data['SkillIDs']);
-            $newSkillIds = array_diff($data['SkillIDs'], $originalSkillIds);
-
-            foreach ($newSkillIds as $newSkill) {
-                if (!$taskSkill = TaskSkill::getByWhere(['TaskID' => $Record->Task->ID, 'SkillID' => $newSkill])) { // check if skill is attached to related task first
-                    StudentTaskSkill::create([
-                        'StudentTaskID' => $Record->ID,
-                        'SkillID' => $newSkill
-                    ], true);
+                if ($Attachment instanceof Attachments\GoogleDriveFile) {
+                    if (!$Attachment->File) {
+                        if (!$File = \Google\DriveFile::getByWhere(['DriveID' => $attachmentData['File']['DriveID']])) {

```

Comment:
@nbey `getByField` would be more appropriate here and get to benefit from caching